### ISSUE TITLE
Function to split training data and its labels into two sets.

### ISSUE THREAD

**Issue by royson** [owlbarn/owl#258]
Would be useful to split arrays and matrices into random train and validation/test subsets similar to 
http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html

**Comment by ryanrhymes** [comment]
I agree, even though this can be done in Owl by first shuffling then slicing, it makes sense to have a more efficient implementation to avoid unnecessary passes and memory copy.
